django/db/models/sql/query.py

```diff
@@ -491,14 +491,19 @@ def get_aggregation(self, using, added_aggregate_names):
             self.default_cols = False
             self.extra = {}
 
+        empty_aggregate_result = [
+            expression.empty_aggregate_value
+            for expression in outer_query.annotation_select.values()
+        ]
+        elide_empty = not any(result is NotImplemented for result in empty_aggregate_result)
```

Comment:
I'm a bit confused, but how it will work for multiple aggregations :thinking: ? e.g.
```python
Book.objects.none().aggregate(pages_count=Count('pages'), rating_sum=Sum('rating'))
```

Reply:
@felixxm the idea is that if any aggregate doesn't not implement `empty_aggregate_value` then the query **must** be run against the database to retrieve the proper results against an empty result set.

In your particular example, since `empty_aggregate_value` is implement for both `Count` and `Sum`, no queries need to be performed.

Reply:
Thanks for explanation. Can we add tests for this behavior?
- multiple aggregations which implement `empty_aggregate_value`,
- multiple aggregations, where one of them doesn't implement `empty_aggregate_value`.